src/portable/portablectl.c

```diff
@@ -717,14 +769,30 @@ static int attach_reattach_image(int argc, char *argv[], const char *method) {
         if (r < 0)
                 return bus_log_create_error(r);
 
+        r = attach_extensions_to_message(m, arg_extension_images);
+        if (r < 0)
+                return r;
+
         r = sd_bus_message_append_strv(m, matches);
         if (r < 0)
                 return bus_log_create_error(r);
 
-        r = sd_bus_message_append(m, "sbs", arg_profile, arg_runtime, arg_copy_mode);
+        r = sd_bus_message_append(m, "s", arg_profile);
         if (r < 0)
                 return bus_log_create_error(r);
 
+        if (STR_IN_SET(method, "AttachImageWithExtensions", "ReattachImageWithExtensions")) {
+                uint64_t flags = arg_runtime ? PORTABLE_RUNTIME : 0;
+
+                r = sd_bus_message_append(m, "st", arg_copy_mode, flags);
+                if (r < 0)
+                        return bus_log_create_error(r);
+        } else {
+                r = sd_bus_message_append(m, "bs", arg_runtime, arg_copy_mode);
+                if (r < 0)
+                        return bus_log_create_error(r);
```

Comment:
doesn't matter but you could move these two lines after the big if block, since both if branches have the same